File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithModuleOf.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

@KoraApp
public interface AppWithModuleOf {
    @Root
    default Class1 class1(Class2 class2) {
        return new Class1(class2);
    }

    record Class1(Class2 class2) {}

    record Class2() {}

    record Class3() {}

    default Class3 class2() {
        return new Class3();
    }

    @ru.tinkoff.kora.common.Module
    interface Module {
        default Class2 class2() {
            return new Class2();
        }
    }

}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithModuleOf.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root

@KoraApp
interface AppWithModuleOf {
    @Root
    fun class1(class2: Class2): Class1 {
        return Class1(class2)
    }

    fun class2(): Class3 {
        return Class3()
    }

    data class Class1(val class2: Class2)
    open class Class2
    class Class3

    @ru.tinkoff.kora.common.Module
    interface Module {
        fun class2(): Class2 {
            return Class2()
        }
    }
}
